File: deploy.php
<?php

require 'vendor/autoload.php';

use phpseclib3\Net\SSH2;

$ssh_host = 'your_server_ip';
$ssh_port = 22;
$ssh_user = 'your_ssh_username';
$ssh_password = 'your_ssh_password'; // Your SSH password

$github_username = 'your_github_username';
$github_token = 'your_github_personal_access_token';
$github_repo_url = "https://$github_username:$github_token@github.com/yourusername/yourrepository.git";
$project_directory = '/path/to/your/project/directory';

$commands = [
    "cd $project_directory && git clone $github_repo_url",
    "cd $project_directory && composer install",
    "cd $project_directory && npm install",
    "cd $project_directory && npm run build",
];

try {
    // Initialize SSH connection
    $ssh = new SSH2($ssh_host, $ssh_port);

    // Authenticate using password
    if (!$ssh->login($ssh_user, $ssh_password)) {
        throw new Exception('Login failed');
    }

    // Execute commands
    foreach ($commands as $command) {
        echo "Executing: $command\n";
        $output = $ssh->exec($command);
        echo "OUTPUT:\n$output\n";
    }

} catch (Exception $e) {
    echo 'Error: ' . $e->getMessage() . "\n";
}

File: deploy.py
# pip install paramiko
import paramiko
import time

# SSH configuration
ssh_host = 'your_server_ip'
ssh_port = 22
ssh_user = 'your_ssh_username'
ssh_password = 'your_ssh_password'  # If using password authentication
ssh_key_file = '/path/to/your/private/key'  # If using SSH key authentication

# GitHub and Project configuration
github_repo_url = 'https://github.com/yourusername/yourrepository.git'
project_directory = '/path/to/your/project/directory'

# Commands to run on the server
commands = [
    f'cd {project_directory} && git clone {github_repo_url}',
    f'cd {project_directory} && composer install',
    f'cd {project_directory} && npm install',
    f'cd {project_directory} && npm run build',
]

def run_ssh_commands(commands):
    # Create an SSH client
    ssh = paramiko.SSHClient()
    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())

    # Connect to the server
    try:
        # If using password authentication
        ssh.connect(ssh_host, port=ssh_port, username=ssh_user, password=ssh_password)

        # If using SSH key authentication
        # ssh.connect(ssh_host, port=ssh_port, username=ssh_user, key_filename=ssh_key_file)

        # Execute commands
        for command in commands:
            print(f"Executing: {command}")
            stdin, stdout, stderr = ssh.exec_command(command)
            stdout.channel.recv_exit_status()  # Block until command finishes
            output = stdout.read().decode()
            error = stderr.read().decode()

            if output:
                print(f"OUTPUT:\n{output}")
            if error:
                print(f"ERROR:\n{error}")

    except Exception as e:
        print(f"Failed to connect or execute commands: {e}")

    finally:
        ssh.close()

if __name__ == "__main__":
    run_ssh_commands(commands)
